larray_editor/arraymodel.py

```diff
@@ -15,8 +14,179 @@
+            remainder = self.total_cols - self.cols_loaded
+            items_to_fetch = min(remainder, self.COLS_TO_LOAD)
+            self.beginInsertColumns(QModelIndex(), self.cols_loaded,
+                                    self.cols_loaded + items_to_fetch - 1)
+            self.cols_loaded += items_to_fetch
+            self.endInsertColumns()
+
+    def get_value(self, index):
+        raise NotImplementedError()
+
+    def _compute_rows_cols_loaded(self):
+        # Use paging when the total size, number of rows or number of
+        # columns is too large
+        size = self.total_rows * self.total_cols
+        if size > LARGE_SIZE:
+            self.rows_loaded = min(self.ROWS_TO_LOAD, self.total_rows)
+            self.cols_loaded = min(self.COLS_TO_LOAD, self.total_cols)
+        else:
+            if self.total_rows > LARGE_NROWS:
+                self.rows_loaded = self.ROWS_TO_LOAD
+            else:
+                self.rows_loaded = self.total_rows
+            if self.total_cols > LARGE_COLS:
+                self.cols_loaded = self.COLS_TO_LOAD
+            else:
+                self.cols_loaded = self.total_cols
+
+    def flags(self, index):
+        raise NotImplementedError()
+
+    def headerData(self, section, orientation, role=Qt.DisplayRole):
+        return to_qvariant()
+
+    def data(self, index, role=Qt.DisplayRole):
+        raise NotImplementedError()
+
+    def reset(self):
+        self.beginResetModel()
+        self.endResetModel()
+
+
+class LabelsArrayModel(AbstractArrayModel):
+    """Labels Table Model.
+
+    Parameters
+    ----------
+    parent : QWidget, optional
+        Parent Widget.
+    data : array-like, optional
+        Input array that can be converted into a LArray 
+        (Numpy ndarray, Pandas Dataframe, list or tuple, ...).
+    readonly : bool, optional
+        If True, data cannot be changed. False by default.
+    font : QFont, optional
+        Font. Default is `Calibri` with size 11.
+    """
+    def __init__(self, parent=None, data=None, readonly=False, font=None):
+        AbstractArrayModel.__init__(self, parent, data, readonly, font)
+        self.font.setBold(True)
+
+    def _set_data(self, data, changes=None):
+        if data is None:
+            data = [[]]
+        if not isinstance(data, (list, tuple)):
+            QMessageBox.critical(self.dialog, "Error", "Expected list or tuple.")
+            data = [[]]
+        self._data = data
+        self.total_rows = len(data[0])
+        self.total_cols = len(data)
+        self._compute_rows_cols_loaded()
+
+    def flags(self, index):
+        """Set editable flag"""
+        return Qt.ItemIsEnabled
+
+    def get_value(self, index):
+        i = index.row()
+        j = index.column()
+        # we need to inverse column and row because of the way ylabels are generated
+        return str(self._data[j][i])
+
+    # XXX: I wonder if we shouldn't return a 2D Numpy array of strings?
```

Comment:
nevermind what I said earlier, I thought this was somewhere else. Unsure about this in this case.